internal/gh/gh.go

```diff
@@ -26,12 +28,31 @@ func CreatePullRequest(args ...string) error {
 	return nil
 }
 
-func ensureGHInstalled() error {
+func EnsureInstalledAndAuthorized() error {
 	cmd := exec.Command("which", "gh")
 	err := cmd.Run()
 	if err != nil {
 		fmt.Println("🤓 This command requires the gh cli.\nSee: https://github.com/cli/cli")
 		return errors.New("missing gh cli dependency")
 	}
+
+	cmd = exec.Command("gh", "auth", "status")
+	output, _ := cmd.CombinedOutput()
+	outputStr := string(output)
+
+	// Check if authorized
+	tokenRegex := regexp.MustCompile(`Token: .+`)
+	if !tokenRegex.MatchString(outputStr) {
+		fmt.Printf("🔐 Please run %s to authorize the gh cli.\n", style.Code("gh auth login"))
```

Comment:
`auth` in the context of the `gh` tool is actually authenticate (verifying who you are). Authorization is checked when you try to do operations (e.g.: trying to clone a repo that you don't have permission to access).